### PR TITLE
Double print label fix

### PR DESCRIPTION
**Pull request by jiphan**
Bar label now only prints once
Also refactored newD3CompHelper.js > createChart()
Also commented out old console.logs

### CODE REVIEW

**Review by crondondev — APPROVED** [review]
LGTM

File: ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js [hunk 1]
```diff
@@ -198,18 +198,18 @@ myAction : function(component, event, helper) {
         if(activePoints.length > 0 ){
             var currIndex = activePoints[0]._index;                
             var currSimpleTraining = component.get('v.tempList')[currIndex];
-            console.log('here is the current simple training: ');
-            console.log(currSimpleTraining.trainingId);
+            //console.log('here is the current simple training: ');
```
Comment (mruizrevature) [inline ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js]: Take out the commented code please.

File: ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js [hunk 2]
```diff
@@ -246,12 +250,15 @@ myAction : function(component, event, helper) {
...
                 this.data.datasets.forEach(function (dataset, i) {
                     var meta = chartInstance.controller.getDatasetMeta(i);
+                    // console.log(meta.data[0]);
                     meta.data.forEach(function (bar, index) {
-                        var model = dataset._meta[Object.keys(dataset._meta)[0]].data[index]._model;                            
-                        var label = model.label;
-                        
-                        ctx.fillText(label, bar._model.x+2, bar._model.y);
-                        
+                        if (bar._datasetIndex === 0) {
```
Comment (mruizrevature) [inline ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js:248]: What's this doing?

File: ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js [hunk 3]
```diff
@@ -343,15 +350,20 @@ myAction : function(component, event, helper) {
             helper.setInitFilterValues(component, event);
             //pass the results to the chart creator.
             var ctx = component.find("myChart").getElement();
-            var newChart = helper.createChart(ctx, barOptions_stacked, response.getReturnValue(),
-                                                component.get('v.UserColors'),  component.find('select').get('v.value'));
+            var newChart = helper.createChart(
```
Comment (mruizrevature) [inline ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js:347]: What's this doing?

**Review by mruizrevature — CHANGES_REQUESTED** [review]
Please remove commented code.

File: ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js (hunk 1 above)
Comment (jiphan) [reply ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js]: done

File: ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js (hunk 2 above)
Comment (jiphan) [reply ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js:248]: will only fillText label to the right of the first invisible bar, added comment for clarification

File: ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js (hunk 3 above)
Comment (jiphan) [reply ProjectForce/force-app/main/default/aura/newD3Comp/newD3CompController.js:347]: put params on different lines for readibility

**Review by mruizrevature — APPROVED** [review]

**Pull request merged**